src/de/grobox/liberario/fragments/ProductDialogFragment.java

```diff
@@ -183,6 +186,13 @@ public void bindView(final ViewHolder ui) {
 				public void onClick(View v) {
 					int position = adapter.getAdapterPosition(ProductItem.this);
 					adapter.toggleSelection(position);
+					EnumSet<Product> products = getProductsFromItems(adapter.getSelectedItems());
```

Comment:
What doesn't work? You would of course not use `EnumSet<Product>` but whatever else the adapter returns. You don't care about the type, just about how many it is, right?